### PR TITLE
CDAP-527: update SparkPageRank program to read directly from stream

### PR DESCRIPTION
**Pull request by artemmakogon**
This PR does not contains changes in docs. They will be added in separate PR

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
:hand: Let's not merge this to develop. We should merge this during release of 2.6.0

File: pom.xml [hunk 1]
```diff
@@ -66,6 +67,11 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
+      <groupId>org.apache.hadoop</groupId>
```
Comment (rohitsinha54) [inline pom.xml:70]: Is this really needed ? 
It should work with provided scope I guess.

**Comment by rohitsinha54** [comment]
Except the import scope everything LGTM

File: pom.xml (hunk 1 above)
Comment (artemmakogon) [inline pom.xml:70]: @rohitsinha5, I have added scope to dependency

**Comment by rohitsinha54** [comment]
@artemmakogon : Have you tested the app after scope change ?

**Comment by artemmakogon** [comment]
@rohitsinha54 yes, I have tested it

**Comment by rohitsinha54** [comment]
@artemmakogon : A lot of changed in class loading on develop. Can you please test again and if everything works go ahead and merge.

**Comment by artemmakogon** [comment]
@rohitsinha54 successfully tested

**Pull request merged**